src/logs/logs_sender.py

```diff
@@ -92,38 +109,40 @@ def prepare_serialized_batches(logs: List[Dict]) -> List[str]:
 
     if len(logs_for_next_batch) >= 1:
         # finalize last batch
-        batch = "[" + ",".join(logs_for_next_batch) + "]"
-        batches.append(batch)
+        serialized_batch = "[" + ",".join(logs_for_next_batch) + "]"
+        batches.append(Batch(serialized_batch, new_batch_len, len(logs_for_next_batch)))
 
     return batches
 
 
-def ensure_fields_length(log_entry):
+def ensure_fields_length(log_entry, context):
     for key, value in log_entry.items():
         if key == "content":
-            ensure_content_length(log_entry)
+            ensure_content_length(log_entry, context)
         elif key == "severity":
             pass
         elif key == "timestamp":
             pass
         else:
-            ensure_attribute_length(log_entry, key, value)
+            ensure_attribute_length(log_entry, key, value, context)
 
 
-def ensure_content_length(log_entry):
+def ensure_content_length(log_entry, context):
     log_content = log_entry.get("content", "")
     log_content_len = len(log_content)
     if log_content_len > DYNATRACE_LOG_INGEST_CONTENT_MAX_LENGTH:
         # placeholder for APM-290579 Implement self-monitoring
         # dynatrace-azure-logs-ingest/logs_ingest/main.py:117
         # self_monitoring.too_long_content_size.append(log_content_len)
 
+        context.sfm.log_content_trimmed()
```

Comment:
Please clean the placeholder above.

+ i'd rather report what was the original length that bool if required trimming. It seems more informative to me, e.g., if it's a problem for the customer, he might be able to manipulate the limit in Dynatrace cluster. 
(Admittedly, there is little chance of this happening unless we document it, but it still seems a step in the right direction)

Reply:
Would you like to sum the length of trimmed logs when sending a metric datapoint to CW? Because then I see no added value for the customer. If not, how would we report it to CloudWatch?

Reply:
Well, then we should report it *in addition* to the number of trimmed logs because it's also important.

Anyway, I think we can add it in the future if customers ask for it (similarly to how we added region reporting in metric streaming when the need arrived) but right now I'm against prolonging the work on this PR, especially since the comment comes two days after Marek initially presented it to us. I think it's asking a bit too much to add non-essential stuff after such a long period